console/console-init/ui/src/modules/iot-device/utils/TableFormatter.tsx

```diff
@@ -1,27 +1,35 @@
 import React from "react";
 import {
   CheckCircleIcon,
-  ExclamationCircleIcon
+  ExclamationCircleIcon,
+  ExclamationTriangleIcon
 } from "@patternfly/react-icons";
 import { Link } from "react-router-dom";
 import { FormatDistance } from "use-patternfly";
 
 import { IRowData } from "@patternfly/react-table";
 import { IDevice } from "modules/iot-device/components";
+import { Label } from "@patternfly/react-core";
 
 export const getTableCells = (row: IDevice) => {
-  const { jsonData = "{}", enabled } = row;
-
-  const { gateways, credentials } = JSON.parse(jsonData);
+  const {
+    enabled,
+    viaGateway,
+    credentials,
+    deviceId,
+    updated,
+    created,
+    lastSeen
+  } = row;
 
   let deviceType: string;
 
-  if (!gateways?.length === !credentials?.length) {
+  if (!viaGateway === !credentials?.length) {
```

Comment:
This check looks confusing. you are comparing boolean and numeric value. Please test this check with all the possible cases. I would like to suggest keep this N/A case as default if possible.

Reply:
This line behaves like XNOR Gate.
![Screenshot from 2020-07-13 17-30-42](https://user-images.githubusercontent.com/23582438/87302471-197d5380-c52f-11ea-977e-bd716be9e4e7.png)